packages/schematics/package-update/_angular-devkit/index.ts

```diff
@@ -0,0 +1,22 @@
+/**
+ * @license
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import { Rule } from '@angular-devkit/schematics';
+import { SchematicsUpdateSchema } from '../schema';
+import { updatePackageJson } from '../utility/npm';
+
+
+const angularPackagesName = [
```

Comment:
Can these always be updated to `@latest` and be compatible together? I don't think they can all be updated to a specific version (e.g. `0.1.2`) at least.

Reply:
They should all be to latest. I don't think we should talk about Angular 2, so I added a check that the version is at least 4.

Reply:
These are actually the devkit (the variable should probably be `devkitPackagesName`). So even though they can be latest, they won't be able to be all at say `0.0.40` (that version exists for `@angular-devkit/schematics` but not `@angular-devkit/build-optimizer` for instance. 

`@angular-devkit/core` is an odd one to be in this list as well, since it's a dependency of the others and you'd probably not want to list it manually.

Reply:
Yes, that's right. I've renamed the variable.

Those are automatic in the sense that you can run `ng update @angular-devkit` and it should work like you expect it (using latest). It's true that this wouldn't work for the devkit, since versions are not fixed. I'd like to fix that in the future, probably, maybe?

I've changed the devkit one to only use a tag.